### PR TITLE
Fixed generation of Geant4 module file and clean-up:

### PR DESCRIPTION
**Pull request by ihrivnac**
- Set "not-defined" value for data sets which are not valid for the built Geant4 version.
  This fixes a failure of the alienv call for older Geant4 versions where some of new data sets are not present.
- Removed setting of data set environment variables from env: section, as they are not used for the build.

### CODE REVIEW

File: geant4.sh [hunk 1]
```diff
@@ -60,18 +47,18 @@ make install
...
-G4LEDATA=`find ${INSTALLROOT} $G4DATASEARCHOPT '*data*G4EMLOW*'`
-G4LEVELGAMMADATA=`find ${INSTALLROOT} $G4DATASEARCHOPT '*data*PhotonEvaporation*'`
-G4NEUTRONHPDATA=`find ${INSTALLROOT} $G4DATASEARCHOPT '*data*G4NDL*'`
-G4NEUTRONXSDATA=`find ${INSTALLROOT} $G4DATASEARCHOPT '*data*G4NEUTRONXS*'`   ## v10.4.px only
-G4PARTICLEXSDATA=`find ${INSTALLROOT} $G4DATASEARCHOPT '*data*G4PARTICLEXS*'`  ## v10.5.px only
-G4PIIDATA=`find ${INSTALLROOT} $G4DATASEARCHOPT '*data*G4PII*'`
-G4RADIOACTIVEDATA=`find ${INSTALLROOT} $G4DATASEARCHOPT '*data*RadioactiveDecay*'`
-G4REALSURFACEDATA=`find ${INSTALLROOT} $G4DATASEARCHOPT '*data*RealSurface*'`
-G4SAIDXSDATA=`find ${INSTALLROOT} $G4DATASEARCHOPT  '*data*G4SAIDDATA*'`
+G4ABLADATA=`find ${INSTALLROOT} $G4DATASEARCHOPT '*data*G4ABLA*'`;[[ -z $G4ABLADATA ]] && G4ABLADATA=`echo  "not-defined"`  ## v10.4.px only
```
Comment (ktf) [inline geant4.sh:50]: Why not simply `G4ABLADATA=not-defined`?

Comment (ihrivnac) [reply geant4.sh]: Yes, thanks, I updated the branch.

File: geant4.sh [hunk 2]
```diff
@@ -14,19 +14,6 @@ incremental_recipe: |
...
-  G4INCLDATA : "`find ${G4INSTALL} $G4DATASEARCHOPT '*data*G4INCL*'`"  ## v10.5.px only
-  G4LEDATA : "`find ${G4INSTALL} $G4DATASEARCHOPT '*data*G4EMLOW*'`"
-  G4LEVELGAMMADATA : "`find ${G4INSTALL} $G4DATASEARCHOPT '*data*PhotonEvaporation*'`"
-  G4NEUTRONHPDATA : "`find ${G4INSTALL} $G4DATASEARCHOPT '*data*G4NDL*'`"
-  G4NEUTRONXSDATA : "`find ${G4INSTALL} $G4DATASEARCHOPT '*data*G4NEUTRONXS*'`"   ## v10.4.px only
-  G4PARTICLEXSDATA : "`find ${G4INSTALL} $G4DATASEARCHOPT '*data*G4PARTICLEXS*'`"   ## v10.5.px only
-  G4PIIDATA : "`find ${G4INSTALL} $G4DATASEARCHOPT '*data*G4PII*'`"
-  G4RADIOACTIVEDATA : "`find ${G4INSTALL} $G4DATASEARCHOPT '*data*RadioactiveDecay*'`"
-  G4REALSURFACEDATA : "`find ${G4INSTALL} $G4DATASEARCHOPT '*data*RealSurface*'`"
-  G4SAIDXSDATA : "`find ${G4INSTALL} $G4DATASEARCHOPT  '*data*G4SAIDDATA*'`"
```
Comment (ktf) [inline geant4.sh:29]: I think we actually need to keep these. Are they needed to build dependent packages?

Comment (ktf) [reply geant4.sh:29]: Ok... Apparently passes the tests, so they are not really needed. Merging.

**Pull request merged**